packages/ember-glimmer/tests/utils/shared-conditional-tests.js

```diff
@@ -374,11 +384,56 @@ export class SharedConditionalsTest extends AbstractConditionalsTest {
     this.assertText('T1F2');
   }
 
+  ['@test it maintains DOM stability when condition changes from a truthy to a different truthy value']() {
+    this.renderValues(true);
+
+    this.assertText('T1');
+
+    this.takeSnapshot();
+
+    this.runTask(() => set(this.context, 'cond1', 'hello'));
+
+    this.assertText('T1');
+
+    this.assertInvariants();
+  }
+
+  ['@test it maintains DOM stability when condition changes from a falsy to a different falsy value']() {
+    this.renderValues(false);
+
+    this.assertText('F1');
+
+    this.takeSnapshot();
+
+    this.runTask(() => set(this.context, 'cond1', ''));
+
+    this.assertText('F1');
+
+    this.assertInvariants();
+  }
+}
+
+export class SharedHelperConditionalsTest extends SharedConditionalsTest {
+  renderValues(...values) {
+    let templates = [];
+    let context = {};
+
+    for (let i = 1; i <= values.length; i++) {
+      templates.push(this.templateFor({ cond: `cond${i}`, truthy: `t${i}`, falsy: `f${i}` }));
```

Comment:
I've gone through the ad-hoc tests, implemented your comments and removed tests which where covered in the shared cases. There is only one ad-hoc one remaining, `it can omit the falsy argument`. My feeling is that it's fine for us to just test a single variation in this case, the shared test cases are fully exercising the truthy and falsy scenarios and the `it can omit the falsy argument` test verifies the behaviour of when the last parameter is omitted.